File: src/react_app/src/interface/admin/EmailLog.ts
export type EmailLogStatus = 'queued' | 'sent' | 'failed' | string;

export interface EmailLogSearchParam {
  keyword?: string;
  status?: string;
  templateCode?: string;
  fromDate?: string;
  toDate?: string;
  limit?: number;
}

export interface EmailLogListItem {
  emailLogSeq: number;
  templateSeq?: number;
  templateCode?: string;
  templateName?: string;
  recipientEmail: string;
  recipientName?: string;
  subject: string;
  status: EmailLogStatus;
  errorMessage?: string;
  retryCount?: number;
  sentAt?: string;
  openedAt?: string;
  createdAt?: string;
}

export interface EmailLogDetail extends EmailLogListItem {
  templateSeq?: number;
  registrationSeq?: number;
  bodyHtml: string;
}

export interface EmailLogResendParam {
  emailLogSeqs: number[];
}

export interface EmailLogResendResult {
  requestedCount: number;
  successCount: number;
  failedCount: number;
}

File: src/main/java/com/kcabEvent/domain/EmailLog.java
package com.kcabEvent.domain;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class EmailLog {
    private Long emailLogSeq;
    private Long templateSeq;
    private String recipientEmail;
    private String recipientName;
    private String subject;
    private String bodyHtml;
    private String status;
    private String errorMessage;
    private Integer retryCount;

    @Builder
    public EmailLog(
            Long templateSeq,
            String recipientEmail,
            String recipientName,
            String subject,
            String bodyHtml,
            String status
    ) {
        this.templateSeq = templateSeq;
        this.recipientEmail = recipientEmail;
        this.recipientName = recipientName;
        this.subject = subject;
        this.bodyHtml = bodyHtml;
        this.status = status;
    }
}
